packages/indexer/src/plugins/persistence.test.ts

```diff
@@ -0,0 +1,61 @@
+import type { Cursor } from "@apibara/protocol";
+import { type Database, open } from "sqlite";
+import sqlite3 from "sqlite3";
+import { afterAll, beforeAll, describe, expect, it } from "vitest";
+import { SqlitePersistence } from "./persistence";
+
+describe("Sqlite Persistence", () => {
```

Comment:
One issue I have with these tests is that they're very dependent on the order in which they're run. 

I would change them to have only 2 cases:

Test 1: assert there's no data, then insert value, check that value was stored, update value, check that value was updated

Test 2: assert there's no data, insert value, check value was stored, delete value, check there's no data.

For each test, you should create a db/persistence instance specific to that test. That way if in the future we run tests in parallel or if vitest changes how they order tests, test won't break.